packages/xo-web/src/xo-app/hub/resource.js

```diff
@@ -0,0 +1,258 @@
+import { find, filter } from 'lodash'
+import { injectState, provideState } from 'reaclette'
+import { withRouter } from 'react-router'
+
+import ResourceForm from './resource-form'
+
+const subscribeAlert = () =>
+  alert(
+    _('hubResourceAlert'),
+    <div>
+      <p>
+        {_('considerSubscribe', {
+          link: 'https://xen-orchestra.com',
+        })}
+      </p>
+    </div>
+  )
+
+export default decorate([
+  withRouter,
+  connectStore(() => {
+    const getTemplates = createGetObjectsOfType('VM-template').sort()
+    const getPools = createGetObjectsOfType('pool')
+    return {
+      templates: getTemplates,
+      pools: getPools,
+      hubInstallLoadingState: state => state.hubInstallLoadingState,
+    }
+  }),
+  provideState({
+    initialState: () => ({
+      selectedInstallPools: [],
+    }),
+    effects: {
+      async install() {
+        const {
+          id,
+          name,
+          namespace,
+          setHubInstallLoadingState,
+          version,
+        } = this.props
+        const { hubInstallLoadingState, isTemplateInstalled } = this.state
+        if (getXoaPlan(process.env.XOA_PLAN) === 'Community') {
+          subscribeAlert()
+          return
+        }
+        const resourceParams = await form({
+          render: props => (
+            <ResourceForm
+              install
+              multi
+              poolPredicate={isTemplateInstalled}
+              {...props}
+            />
+          ),
+          header: (
+            <span>
+              <Icon icon='add-vm' /> {name}
+            </span>
+          ),
+          size: 'medium',
+        })
+
+        setHubInstallLoadingState({
+          ...hubInstallLoadingState,
+          [id]: true,
+        })
+        try {
+          await Promise.all(
+            resourceParams.pools.map(pool =>
+              downloadAndInstallResource({
+                namespace,
+                id,
+                version,
+                sr: pool.default_SR,
+              })
+            )
+          )
+          success('XVA import', _('successfulInstall'))
+        } catch (_error) {
+          error('Error', _error.message)
+        }
+        setHubInstallLoadingState({
+          ...hubInstallLoadingState,
+          [id]: false,
+        })
+      },
+      async create() {
+        const { isPoolCreated, installedTemplates } = this.state
+        const { name } = this.props
+        if (getXoaPlan(process.env.XOA_PLAN) === 'Community') {
+          subscribeAlert()
+          return
+        }
+        const resourceParams = await form({
+          render: props => (
+            <ResourceForm poolPredicate={isPoolCreated} {...props} />
+          ),
+          header: (
+            <span>
+              <Icon icon='add-vm' /> {name}
+            </span>
+          ),
+          size: 'medium',
+        })
+        const { $pool } = resourceParams.pool
+        const pool = find(installedTemplates, { $pool })
```

Comment:
`pool` → `template`, then?

Reply:
What do you mean ? I don't get it :slightly_smiling_face:

Reply:
`find` returns a template, here, not a pool, right?